src/barnetilsyn/BarnetilsynContext.tsx

```diff
@@ -185,6 +202,43 @@ const [BarnetilsynSøknadProvider, useBarnetilsynSøknad] = createUseContext(
       }
     };
 
+    const overskrivBarnFraForrigeSøknadMedPersonData = (
+      barn: IBarn,
+      personData: PersonData
+    ): IBarn | undefined => {
+      const gjeldendeBarn = personData.barn.find(
+        (personBarn) => personBarn.fnr === barn.ident.verdi
+      );
+
+      if (!gjeldendeBarn) return undefined;
```

Comment:
Hvis gjeldendeBarn ikke finnes her - da gjør vi ingen endringer på "eksisterende" barneprops- sant? 

Er det en ønsket situasjon? I hvilke tilfeller kan dét evt.  skje?

Reply:
Funksjonen returnere undefined, men det blir brukt spread operator på variabelen som holder den verdien. Det er trygt fordi det ikke skjer noe da hvis den er undefined.

```
                  const **barnFraPersonData** =
                    overskrivBarnFraForrigeSøknadMedPersonData(
                      barn,
                      personData
                    );
```
                    
```
                  return {
                    ...barn,
                    ...**barnFraPersonData**,
                    medforelder,
                    forelder: oppdatertForelder,
                    fraFolkeregister: fraFolkeregister,
                    erFraForrigeSøknad: true,
                  };
```

Vi gjør ingen endringer på eksisterende barn.

Reply:
Skjønner. Det jeg egentlig lurer mest på her er - i hvilke situasjoner oppstår dette?
Når du har barn fra forrige søknad som dør, f.eks?

Reply:
Barn som er for gamle eller død blir ikke med fra personData/aktuelleBarn:
```
          const aktuelleBarn = forrigeSøknad.person.barn.filter((barn) =>
            personData.barn.some(
              (personBarn) => personBarn.fnr === barn.ident.verdi
            )
          );
```

find() kan returnere undefined. Sjekken er egentlig med for å håndtere dette slik at vi ikke behøver optional chaining osv.

I teorien bør det derfor aldri bli undefined, kanskje vi heller bør kaste en error?

Reply:
Error i frontend bør for all del unngås - da krasjer jo søknaden :)